electron/src/lib/CertificateVerifyProcManager.ts

```diff
@@ -0,0 +1,184 @@
+import * as path from 'path';
+import {URL} from 'url';
+import * as environment from '../js/environment';
+import {getText} from '../locale/locale';
+import {LogFactory} from '../util/';
+
+const LOG_DIR = path.join(app.getPath('userData'), 'logs');
+LogFactory.LOG_FILE_PATH = LOG_DIR;
+LogFactory.LOG_FILE_NAME = 'electron.log';
+
+const logger = LogFactory.getLogger('CertificateVerifyProcManager.ts', {forceEnable: true});
+
+interface DisplayCertificateErrorOptions {
+  bypassDialogLock: boolean;
+  isChromiumError: boolean;
+  isCheckboxChecked: boolean;
+}
+
+class CertificateVerifyProcManager {
+  private static bypassCertificatePinning: boolean = false;
+  private static isDialogLocked: boolean = false;
+
+  public static readonly CHROMIUM_ERRORS = {
+    CERT_AUTHORITY_INVALID: -202,
+    CERT_COMMON_NAME_INVALID: -200,
+  };
+  private static readonly RESPONSE = {
+    RETRY: 0,
+    SHOW_DETAILS: 1,
+  };
+
+  private static displayCertificateDetails(
+    hostname: string,
+    certificate: Electron.Certificate,
+    options: DisplayCertificateErrorOptions
+  ) {
+    dialog.showCertificateTrustDialog(
+      {
+        certificate,
+        message: `${
+          options.isChromiumError
+            ? getText('certificateVerifyProcManagerShowDetailsTextChromium')
+            : getText('certificateVerifyProcManagerShowDetailsTextPinning')
+        } ${hostname}`,
+      },
+      () => {
+        // Go back to the dialog
+        this.displayCertificateError(hostname, certificate, {
+          ...options,
+          bypassDialogLock: true,
+        });
+      }
+    );
+  }
+
+  public static isCertificatePinningEnabled() {
+    return !this.bypassCertificatePinning;
+  }
+  public static displayCertificateChromiumError(hostname: string, certificate: Electron.Certificate) {
+    this.displayCertificateError(hostname, certificate, {isChromiumError: true});
+  }
+
+  public static displayCertificateError(
+    hostname: string,
+    certificate: Electron.Certificate,
+    options?: Partial<DisplayCertificateErrorOptions>
+  ) {
+    const {bypassDialogLock, isChromiumError, isCheckboxChecked} = {
+      bypassDialogLock: false,
+      isCheckboxChecked: false,
+      isChromiumError: false,
+      ...options,
+    };
+    if (this.isDialogLocked && !bypassDialogLock) {
+      return;
+    }
+    this.isDialogLocked = true;
+
+    dialog.showMessageBox(
+      {
+        buttons: [getText('certificateVerifyProcManagerRetry'), getText('certificateVerifyProcManagerShowDetails')],
+        checkboxChecked: isChromiumError ? undefined : isCheckboxChecked,
+        checkboxLabel: isChromiumError ? undefined : getText('certificateVerifyProcManagerWarningBypass'),
+        detail: isChromiumError
+          ? getText('certificateVerifyProcManagerWarningTextChromium')
+          : getText('certificateVerifyProcManagerWarningTextPinning'),
+        message: getText('certificateVerifyProcManagerWarningTitle'),
+        type: 'warning',
```

Comment:
Please create a few constants here instead of moving it all into one line. This is not well readable.